pkg/validation/validation.go

```diff
@@ -112,6 +122,49 @@ func NodePorts(contour *operatorv1alpha1.Contour) error {
 	return nil
 }
 
+// loadBalancer validates LoadBalancer "address" parameter of contour, returning an
+// error if "address" does not meet the API specification.
+func LoadBalancerAddress(contour *operatorv1alpha1.Contour) error {
+	if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.AzureLoadBalancerProvider &&
+		contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure != nil &&
+		contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Address != nil {
+		validationIP := net.ParseIP(*contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Address)
+		if validationIP == nil {
+			return fmt.Errorf("wrong LoadBalancer address format, should be string with IPv4 or IPv6 format")
+		}
+	} else if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.GCPLoadBalancerProvider &&
+		contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP != nil &&
+		contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP.Address != nil {
+		validationIP := net.ParseIP(*contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP.Address)
+		if validationIP == nil {
+			return fmt.Errorf("wrong LoadBalancer address format, should be string with IPv4 or IPv6 format")
+		}
+	}
+
+	return nil
+}
+
+func LoadBalancerProvider(contour *operatorv1alpha1.Contour) error {
+	if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.AWSLoadBalancerProvider {
+		if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure != nil ||
+			contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP != nil {
+			return fmt.Errorf("aws provider chosen, other providers parameters should not be specified")
+		}
+	} else if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.AzureLoadBalancerProvider {
+		if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.AWS != nil ||
+			contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP != nil {
+			return fmt.Errorf("azure provider chosen, other providers parameters should not be specified")
+		}
+	} else if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.GCPLoadBalancerProvider {
+		if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.AWS != nil ||
+			contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure != nil {
+			return fmt.Errorf("gcp provider chosen, other providers parameters should not be specified")
+		}
+	}
+
+	return nil
+}
```

Comment:
nit: This conditional will be easier to read using a switch on the provider parameters type:
```
switch contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type {
  case operatorv1alpha1.AWSLoadBalancerProvider:
...
}